### PR TITLE
Update day03.md - Discussing the solution

### PR DESCRIPTION
**Pull request by makingthematrix**

### CODE REVIEW

File: docs/2024/puzzles/day03.md [hunk 1]
```diff
@@ -6,6 +6,120 @@ import Solver from "../../../../../website/src/components/Solver.js"
...
+
+The shopkeeper at the North Pole Toboggan Rental Shop asks us to go through the input string, find all substrings that look like calls to a multiplication method (`"mul(a,b)`), perform the multiplications, and sum them together. To find such substrings we will construct a regular expression which will look for the following:
+1. Each substring starts with a text `mul(`. In the regex form we need to escape the parenthesis, so this will be `mul\(`.
+2. The text is followed by the first positive integer. Since it's an integer and it's positive we don't need to look for floating points or minuses, we can just look for sequences of ciphers. In regex, one cipher is denoted by `\d`, and since we look for a sequence of one or more ciphers, it's `\d+`.
+3. After the first number there should be a comma, and in regex that again needs to be escaped, so `\,`.
+4. Then there should be a second number, so agian `\d+`.
+5. And the substring should finish with the closing parentheses, escaped again, so `\)`.
+That's it. Our whole regex looks like this: 
+```scala
+val mulPattern: Regex = """mul\((\d+)\,(\d+)\)""".r
```
Comment (sjrd) [inline docs/2024/puzzles/day03.md:33]: `,` doesn't need to be quoted:
```suggestion
val mulPattern: Regex = """mul\((\d+),(\d+)\)""".r
```
Also, have you considered using `java.util.regex.Pattern.quote` or the `\Q...\E` escapes? They might not do much good for `mul`, but for `do()` and `don't()` that would probably make it easier to read.

Comment (makingthematrix) [reply docs/2024/puzzles/day03.md]: I made one more commit, removing the part about escaping commas.

File: docs/2024/puzzles/day03.md
Comment (KacperFKorban) [inline docs/2024/puzzles/day03.md:1]: Could you add yourself as the author of the solution, as with the first one? https://scalacenter.github.io/scala-advent-of-code/2024/puzzles/day01

**Review by KacperFKorban — APPROVED** [review]
Looks good to me!

File: docs/2024/puzzles/day03.md
Comment (makingthematrix) [reply docs/2024/puzzles/day03.md:1]: I added my solution in a separate commit. It's already merged.

Comment (SethTisue) [reply docs/2024/puzzles/day03.md:1]: (I think #590 is what Kacper meant)

**Pull request merged**